core/domain/exp_services.py

```diff
@@ -1998,28 +2134,93 @@ def update_exploration(
+        )
+    )
 
+    user_data_model_to_put = get_user_data_model_with_draft_discarded(
+        exploration_id, committer_id)
+    if user_data_model_to_put:
+        models_to_put.append(user_data_model_to_put)
     if committer_id != feconf.MIGRATION_BOT_USER_ID:
-        user_services.add_edited_exploration_id(committer_id, exploration_id)
-        user_services.record_user_edited_an_exploration(committer_id)
-        if not rights_manager.is_exploration_private(exploration_id):
-            user_services.update_first_contribution_msec_if_not_set(
-                committer_id, utils.get_current_time_in_millisecs())
+        models_to_put.extend(
+            user_services.add_edited_exploration_id_to_model(
+                committer_id,
+                exploration_id
+            )
+        )
+        user_settings_model = user_services.get_user_settings(
+            committer_id,
+            strict=False
+        )
+        if user_settings_model:
+            models_to_put.append(
+                user_services.record_user_edited_an_exploration_in_model(
+                    committer_id
+                )
+            )
+            if not rights_manager.is_exploration_private(exploration_id):
+                updated_user_settings = (
+                    user_services
+                    .update_first_contribution_msec_if_not_set_in_model(
+                        committer_id,
+                        utils.get_current_time_in_millisecs()
+                    )
+                )
+                if updated_user_settings:
+                    user_settings_model = (
+                        user_services.convert_to_user_settings_model(
+                            updated_user_settings
+                        )
+                    )
+                    models_to_put.append(user_settings_model)
 
     if opportunity_services.is_exploration_available_for_contribution(
             exploration_id):
-        opportunity_services.update_opportunity_with_updated_exploration(
-            exploration_id)
+        models_to_put.extend(
+            opportunity_services
+            .update_opportunity_models_with_updated_exploration(
+                exploration_id
+            )
+        )
+
+    if is_synchronous:
+        exp_rights = exp_models.ExplorationRightsModel.get_by_id(
+            exploration_id
+        )
+        exp_summary_model = exp_models.ExpSummaryModel.get(exploration_id)
+        exp_summary = update_exploration_summary(
+            updated_exploration,
+            exp_rights,
+            exp_fetchers.get_exploration_summary_from_model(exp_summary_model),
+            skip_exploration_model_last_updated=True
+        )
+        exp_summary.version += 1
+        updated_exp_summary_model: exp_models.ExpSummaryModel = (
+            populate_exp_summary_model_fields(
+                exp_summary_model, exp_summary
+            )
+        )
+        models_to_put.append(updated_exp_summary_model)
```

Comment:
Let's get rid of the if/else branch and the extra argument and make it always synchronous. It's just one extra model, it's not worth the extra complexity here IMO.

Also, when testing in a production server make sure to do some manual runs saving private/public explorations to see whether anything breaks with the put_multi() in normal request/response operation.